haskell-src/Concordium/Types/Tokens.hs

```diff
@@ -9,14 +9,21 @@ import qualified Data.Aeson.Types as AE
 import qualified Data.ByteString.Short as BSS
 import Data.Scientific
 import qualified Data.Serialize as S
+import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 import Data.Word
 
 -- | The unique token identifier for a protocol-level token.
 --  This is given as a symbol unique across the whole chain.
 --  The byte string must be at most 255 bytes long and be a valid UTF-8 string.
 newtype TokenId = TokenId {tokenSymbol :: BSS.ShortByteString}
-    deriving newtype (Eq, Ord, Show)
+    deriving newtype (Eq, Ord)
+
+instance Show TokenId where
+    show (TokenId sbs) =
+        case T.decodeUtf8' (BSS.fromShort sbs) of
+            Right txt -> T.unpack txt
+            Left err -> "TokenId is not valid UTF-8: " ++ show err
```

Comment:
```suggestion
        T.unpack (T.decodeUtf8Lenient (BSS.fromShort sbs))
```
I think [`decodeUtf8Lenient`](https://hackage-content.haskell.org/package/text-2.1.2/docs/Data-Text-Encoding.html#v:decodeUtf8Lenient) makes sense here. In any case it should be valid UTF-8.